src/dos/drive_iso.cpp

```diff
@@ -211,32 +211,35 @@ void isoDrive::Activate(void) {
 	UpdateMscdex(driveLetter, fileName, subUnit);
 }
 
-bool isoDrive::FileOpen(DOS_File **file, const char *name, uint8_t flags) {
+std::unique_ptr<DOS_File> isoDrive::FileOpen(const char* name, uint8_t flags)
+{
 	if ((flags & 0x0f) == OPEN_WRITE) {
 		DOS_SetError(DOSERR_ACCESS_DENIED);
-		return false;
+		return nullptr;
 	}
 
 	isoDirEntry de;
-	bool success = lookup(&de, name) && !IS_DIR(FLAGS1);
-
-	if (success) {
-		FileStat_Block file_stat;
-		file_stat.size = DATA_LENGTH(de);
-		file_stat.attr = FatAttributeFlags::ReadOnly;
-		file_stat.date = DOS_PackDate(1900 + de.dateYear, de.dateMonth, de.dateDay);
-		file_stat.time = DOS_PackTime(de.timeHour, de.timeMin, de.timeSec);
-		*file = new isoFile(this, name, &file_stat, EXTENT_LOCATION(de) * ISO_FRAMESIZE);
-		(*file)->flags = flags;
+	if (!lookup(&de, name) && !IS_DIR(FLAGS1)) {
+		return nullptr;
 	}
-	return success;
+
+	FileStat_Block file_stat;
+	file_stat.size = DATA_LENGTH(de);
+	file_stat.attr = FatAttributeFlags::ReadOnly;
+	file_stat.date = DOS_PackDate(1900 + de.dateYear, de.dateMonth, de.dateDay);
```

Comment:
Maybe extract 1900 to `DosEpochYear` or something?

Reply:
DOS uses 1980 as epoch, at least in the packed format.  1900 comes from C maybe?  https://cplusplus.com/reference/ctime/tm/

I'm not actually sure what it means it this context because it's coming from some internal ISO direntry struct.

Reply:
Yeah, okay, maybe job for another day... 😅